test/integration/framework/etcd.go

```diff
@@ -136,6 +138,27 @@ func startEtcd() (func(), error) {
 	if err := cmd.Start(); err != nil {
 		return nil, fmt.Errorf("failed to run etcd: %v", err)
 	}
+
+	i := 0
+	target := int(300)
+
+	for i < target {
+		conn, err = net.DialTimeout("tcp", strings.TrimPrefix(etcdURL, "http://"), 1*time.Second)
+		if err == nil {
+			conn.Close()
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I would move the `if i == target { stop && return fmt.Errorf()}` check before this line

Reply:
Done. Thanks.